src/thunderbots/software/ai/primitive/primitive_factory.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ *  Implementations for functions to construct Primitives from various sources
+ */
+
+#include "ai/primitive/primitive_factory.h"
+
+#include <exception>
+
+#include "ai/primitive/catch_primitive.h"
+#include "ai/primitive/chip_primitive.h"
+#include "ai/primitive/direct_velocity_primitive.h"
+#include "ai/primitive/directwheels_primitive.h"
+#include "ai/primitive/dribble_primitive.h"
+#include "ai/primitive/kick_primitive.h"
+#include "ai/primitive/move_primitive.h"
+#include "ai/primitive/movespin_primitive.h"
+#include "ai/primitive/pivot_primitive.h"
+#include "ai/primitive/stop_primitive.h"
+
+
+std::unique_ptr<::Primitive> AI::Primitive::createPrimitiveFromROSMessage(
```

Comment:
Is the double colon necessary in front of `Primitive` here? (and everywhere else)?

Reply:
Yup, until we put `Primitive` in the `AI::Primitive` namespace, we need to explicitly state that we want `Primitive` from the global namespace.

Actually, you've pointed out the one case where it looks like we don't technically need to do this, but I'm inclined to leave it as-is right now to be consistent with the other instances.